src/Services/PatientService.php

```diff
@@ -173,6 +176,93 @@ public function databaseInsert($data)
         }
     }
 
+    /**
+     * Grants portal access to patients in case they do
+     * not have the permission.
+     *
+     * @param $puuidString
+     * @return void
+     */
+    public function allowPatientPortal($puuidString): void
+    {
+        $sql = "UPDATE `patient_data` SET";
+        $sql .= " `allow_patient_portal`='YES'";
+        $sql .= " WHERE `uuid` = ?";
+        $puuidBinary = UuidRegistry::uuidToBytes($puuidString);
+        sqlQuery($sql, array($puuidBinary));
+    }
+
+    /**
+     * Verifies if the patient has access to the portal.
+     *
+     * @param $patient_data
+     * @return boolean
+     */
+    public function hasPortalPermission($patient_data): bool
+    {
+        $patient_portal_permission = strtolower($patient_data["allow_patient_portal"]);
+
+        return !(
+            empty($patient_data["allow_patient_portal"]) ||
+            $patient_portal_permission != self::HAS_PORTAL_PERMISSION
+        );
+    }
+
+    /**
+     * It is responsible for saving a patient's credentials.
+     *
+     * @param $patient_data, $patient_password
+     * @return void
+     */
+    private function onSaveCredentials($patient_data, $patient_password): void {
+        $portal_permission = 1;
+        $patientAccessOnSiteService = new PatientAccessOnsiteService(); 
+        $patientAccessOnSiteService->saveCredentials(
+            $patient_data["pid"], 
+            $patient_password, 
+            $patient_data["fname"], 
+            $patient_data["email"], 
+            $portal_permission
+        );
+    }
+
+    /**
+     * This function is responsible for adding a password for
+     * patients to access the portal. If no password is specified,
+     * the default password is the document number.
+     *
+     * @param $patient_data
+     * @return ProcessingResult
+     */
+    public function setPatientAccess($puuidString, $data): ProcessingResult
```

Comment:
Good, now, use the type hinting too for the parameters, not only for the return type:

public function setPatientAccess(string $puuidString, array $data): ProcessingResult